tests/unit/CliftTypes.cpp

```diff
@@ -0,0 +1,51 @@
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "mlir/IR/Diagnostics.h"
+
+#include "revng-c/mlir/Dialect/Clift/IR/Clift.h"
+#include "revng-c/mlir/Dialect/Clift/IR/ImportModel.h"
+
+#include "ModelTypes.inc"
+
+static bool doesVerifyNames() {
+  return false;
+}
+
+template<typename CallableType>
+static auto withContext(CallableType Callable) {
+  mlir::MLIRContext Context;
+  Context.loadDialect<mlir::clift::CliftDialect>();
+
+  const auto EmitError = [&]() -> mlir::InFlightDiagnostic {
+    return Context.getDiagEngine().emit(mlir::UnknownLoc::get(&Context),
+                                        mlir::DiagnosticSeverity::Remark);
+  };
+
+  return Callable(EmitError, Context);
+}
+
+static bool verify(const model::Type &ModelType, const bool Assert) {
+  return withContext([&](const auto EmitError, mlir::MLIRContext &Context) {
+    return static_cast<bool>(revng::getUnqualifiedTypeChecked(EmitError,
+                                                              Context,
+                                                              ModelType));
+  });
+}
+
+static bool verify(const model::QualifiedType &ModelType, bool Assert) {
+  return withContext([&](const auto EmitError, mlir::MLIRContext &Context) {
+    return static_cast<bool>(revng::getQualifiedTypeChecked(EmitError,
+                                                            Context,
+                                                            ModelType));
+  });
+}
+
+static bool verify(const model::Binary &Tree, const bool Assert) {
+  return Tree.verify(false);
+}
+
+static bool checkSerialization(const TupleTree<model::Binary> &Tree) {
+  return true;
+}
```

Comment:
:thinking: I'm puzzled by this. What's the purpose of having a test that always returns `true`? Is this something you plan to finalize in a second moment?

Reply:
This is for sharing the test code between the model and Clift. This code is taken from here: https://github.com/revng/revng/blob/develop/tests/unit/ModelType.cpp
These static functions are "customisations" of the test where different functions are provided in the model and Clift tests. There's no need to test model serialisation again in the Clift test.